### PR TITLE
[#1808] Swipable `VaCarousel`

### PR DESCRIPTION
**Pull request by aluarius**
Closes: #1808 

## Description
Swiping composable + integration to `VaCarousel`.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/composables/useSwipe.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    return result === props.swipeDirection || commonAllowedDirections[direction].includes(props.swipeDirection) ? result : ''
+  }
+
+  const swipeState = reactive({ direction: '', duration: 0 }) as SwipeState
+  watch(swipePath, () => {
+    const xDistance = calcDistance('x')
+    const yDistance = calcDistance('y')
+
+    if ((xDistance || yDistance) && [xDistance, yDistance].some((el) => Math.abs(el) >= props.swipeDistance)) {
+      if (Math.abs(xDistance) > Math.abs(yDistance) && isSwipeAllowed.value.horizontal) {
```
Comment (m0ksem) [inline packages/ui/src/composables/useSwipe.ts:116]: Would be nice to have `reactiveComputed` to change it to isSwipeAllowed.horizontal

File: packages/ui/src/composables/useSwipe.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  const resetSwipe = () => {
+    (['start', 'end'] as SwipePosition[]).forEach((type) => {
+      swipePath[type].x = 0
+      swipePath[type].y = 0
+      swipeDuration[type] = 0
+    })
+    swipeStarted.value = false
+  }
+
+  const isSwipeAllowed = computed(() => ({
```
Comment (m0ksem) [inline packages/ui/src/composables/useSwipe.ts:94]: Or having isHorizontalSwipeAllowed

**Review by m0ksem — APPROVED** [review]
Bruh

Have issue with `user-select`. Would be nice to have `useUnselectable` :D

Also, for me, swipe feels too sensitive... Would be nice if I need to make more effort for swipe. This just unnatural.

File: packages/ui/src/composables/useSwipe.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  const swipeState = reactive({ direction: '', duration: 0 }) as SwipeState
+  watch(swipePath, () => {
+    const xDistance = calcDistance('x')
+    const yDistance = calcDistance('y')
+
+    if ((xDistance || yDistance) && [xDistance, yDistance].some((el) => Math.abs(el) >= props.swipeDistance)) {
+      if (Math.abs(xDistance) > Math.abs(yDistance) && isSwipeAllowed.horizontal) {
+        const result = xDistance > 0 ? 'left' : 'right'
+        swipeState.direction = getAcceptableValue('horizontal', result)
+      } else if (isSwipeAllowed.vertical) {
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useSwipe.ts:123]: why do we check the condition `Math.abs(xDistance) > Math.abs(yDistance)` on line 120 and don't check the condition on line 123 accordingly?

File: packages/ui/src/composables/useSwipe.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        swipeState.direction = getAcceptableValue('vertical', result)
+      }
+
+      swipeState.duration = swipeDuration.end - swipeDuration.start
+
+      resetSwipe()
+    }
+  }, { deep: true })
+
+  watch(swipeState, cb, { deep: true })
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useSwipe.ts:134]:
```suggestion
  watch(swipeState, () => cb(swipeState.direction), { deep: true })
```

File: packages/ui/src/composables/useSwipe.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+) => {
+  const swipeStarted = ref(false)
+  const swipePath = reactive({
+    start: { x: 0, y: 0 },
+    end: { x: 0, y: 0 },
+  }) as SwipePath
+  const swipeDuration = reactive({
+    start: 0,
+    end: 0,
+  })
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useSwipe.ts:58]: This variable isn't used

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

**Pull request merged**